packages/turbo/ios/RNSession.swift

```diff
@@ -11,61 +11,58 @@ import WebKit
 
 class RNSession: NSObject {
   
-  private var registeredVisitableViews: [SessionSubscriber] = []
-  private var eventEmitter: RCTEventEmitter? = nil
-  private var sessionHandle: NSString = "defaultHandle"
-  private var applicationNameForUserAgent: String? = nil
-    
-  override init() { }
-    
-  init(eventEmitter: RCTEventEmitter, sessionHandle: NSString){
-    self.eventEmitter = eventEmitter
+  private var visitableViews: [SessionSubscriber] = []
+  private var sessionHandle: NSString
+  private var webViewConfiguration: WKWebViewConfiguration
+  
+  init(sessionHandle: NSString, webViewConfiguration: WKWebViewConfiguration){
     self.sessionHandle = sessionHandle
+    self.webViewConfiguration = webViewConfiguration
   }
-    
-  convenience init(eventEmitter: RCTEventEmitter, sessionHandle: NSString, applicationNameForUserAgent: NSString?){
-    self.init(eventEmitter: eventEmitter, sessionHandle: sessionHandle)
-    self.applicationNameForUserAgent = applicationNameForUserAgent as String?
-  }
-  
+
   public lazy var turboSession: Session = {
-    let configuration = WKWebViewConfiguration()
-    configuration.userContentController.add(self, name: "nativeApp")
-    configuration.userContentController.add(RNStradaWKUserContentController(eventEmitter: eventEmitter), name: "stradaNative")
-    configuration.applicationNameForUserAgent = applicationNameForUserAgent
-    return Session(webViewConfiguration: configuration)
+    self.webViewConfiguration.userContentController.add(self, name: "nativeApp")
+    return Session(webViewConfiguration: self.webViewConfiguration)
   }()
   
   func registerVisitableView(newView: SessionSubscriber) {
-    if (!registeredVisitableViews.contains {
+    if (!visitableViews.contains {
       $0.id == newView.id
     }) {
-      registeredVisitableViews.append(newView)
+      visitableViews.append(newView)
     }
-    newView.attachDelegateAndVisit(newView.controller!)
+    newView.attachDelegateAndVisit(newView.controller)
   }
   
   func removeVisitableView(view: SessionSubscriber) {
```

Comment:
I find this method really hard to read and reason about. Maybe we could refactor it for better readability?